esp-hal-common/src/spi.rs

```diff
@@ -337,6 +363,143 @@ mod ehal1 {
             self.spi.flush()
         }
     }
+
+    use core::cell::RefCell;
+
+    use embedded_hal_1::spi::{self, blocking::SpiDevice, ErrorType};
+    #[cfg(any(feature = "esp32", feature = "esp32s2", feature = "esp32s3"))]
+    use xtensa_lx::mutex::Mutex;
+    #[cfg(any(feature = "esp32", feature = "esp32s3"))]
+    use xtensa_lx::mutex::SpinLockMutex as MutexImpl;
+    #[cfg(feature = "esp32s2")]
+    use xtensa_lx::mutex::CriticalSectionMutex as MutexImpl;
+    #[cfg(feature = "esp32c3")]
+    use riscv::interrupt::Mutex as MutexImpl;
```

Comment:
Are we able to use the new `CriticalSection` from the HAL here instead?

Reply:
Maybe? I don't know, I wasn't aware something like that exists. :-)

Looking at the code I wouldn't know what speaks against it. Is that safe to use across multiple cores? How does it differ from the existing Mutexes?

Reply:
Yes, that is quite new and the preferred way now

It's multicore safe on multicore chips. In our case the API is much more convenient than what we had for Xtensa 
For a general introduction to why this is better than what there was before I think this does a great job: https://crates.io/crates/critical-section

Reply:
Got it, implemented it and I agree it's much more convenient!